### DISCUSSION TITLE
how to modify nuxt server start port, in nuxt 3?

### DISCUSSION THREAD

**Discussion by lazercaveman** [nuxt/nuxt#15910] [Questions]
Before it was possible to setup the server object within the nuxt.config - this seems to be deprecated for nuxt 3 - I could not find any information about that process within the docs - so my question is how this should be done in Nuxt 3.

**Comment by Diizzayy** [comment]
@lazercaveman quick answer: use `--port` in your dev script or the `PORT` environment variable

- Dev Server

For the dev server you can simply set the `PORT` or `NUXT_PORT` environment variables

or use the `--port` flag as seen below

```json
{
  "scripts": {
    "dev": "nuxt dev --port 8001"
  }
}
```

- Production Server

For the production server you can simply set the `PORT` or `NITRO_PORT` environment variables

`PORT=8001 node .output/server/index.mjs `

**Reply by praveshishere** [reply]
NUXT_PORT or PORT in the .env file doesn't work for me. it always starts on port 3000, however, adding the following in package.json works
```json
"scripts": {
    "dev": "nuxt dev --port 8001"
}
```

is there something I'm missing to run it on a different port using the .env file

**Reply by danielroe** [reply]
This has been implemented in the edge channel and should be present in RC7: https://github.com/nuxt/framework/pull/6119

**Comment by ElderlyBoy** [comment]
Create an `.env` file in your program's root directory.
````ini
PORT=8080
HOST=0.0.0.0
````
This can change the server running port and host both Vite and Nitro.

**Reply by festusiyere** [reply]
This worked. 
Thanks

**Comment by lucas-cota** [comment]
Does this way also serve to specify the port in production mode?

**Reply by festusiyere** [reply]
Yes it does

**Reply by lucas-cota** [reply]
thank you!

**Comment by elzodxon** [comment]
I don't think, it works. It is not working.

**Reply by festusiyere** [reply]
What version of next you using?

**Reply by elzodxon** [reply]
I am using Nuxt  3:)

**Comment by elzodxon** [comment]
```
module.exports = {
  apps: [
    {
      name: 'NuxtAppName',
      port: '3005',
      exec_mode: 'cluster',
      instances: 'max',
      script: './.output/server/index.mjs'
    }
  ]
}
```

It worked for me.

**Reply by Contentstack-AnkitaD** [reply]
Is this in Nuxt3, NUXT3 uses export default defineNuxtConfig right..

**Reply by jboada** [reply]
Hi @elzodxon ,

I used this solution and it throws a big js module error.

I found a solution on this https://github.com/Unitech/pm2/issues/5337#issuecomment-1094791628 .

It mentions that renaming the ecosystem file from ecosystem.config.js to ecosystem.config.cjs solves the issue. I applied and it worked.

Did you have to rename the file or do any extra config in a file or files?

JB

**Comment by Mansehej** [comment]
You can now define the port in your `nuxt.config` as follows:

```js
export default defineNuxtConfig({
  devServer: {
    port: 3001,
  },
})
```

**Reply by ihteshamqazi** [reply]
problem is production server port, not dev port

**Comment by Bryceed** [comment]
Maybe an update changed the way the production port was configured? I can't switch in any case mentioned earlier.

![image](https://github.com/nuxt/nuxt/assets/42657376/e5c14188-1923-4550-bec0-5750b38c27e5)

It's a new project using the latest version `("nuxt": "^3.7.4")`

**Reply by Bryceed** [reply]
Also, this isn't related with some "wrong message in terminal"

![image](https://github.com/nuxt/nuxt/assets/42657376/4b5d1929-68d9-4785-839c-8a4fba30fd2e)

**Reply by Bryceed** [reply]
I "solved" using this workaround (https://stackoverflow.com/a/72883404), but definitely there's something wrong for production scenario (_or not well updated in the documentation_).

**Reply by PangHao-NEFU** [reply]
you can try cross-env PORT=8080 node.output/xxx/index.mjs

**Comment by Robertv9** [comment]
For me works creating a pm2 file like the docs. example.  [](https://nuxt.com/docs/getting-started/deployment#pm2)
Just change port number  and  In your deployment script add this line "pm2 start ecosystem.config.cjs "
`
module.exports = {
  apps: [
    {
      name: 'NuxtAppName',
      port: '5000',
      exec_mode: 'cluster',
      instances: 'max',
      script: './.output/server/index.mjs'
    }
  ]
}
`

Note: Before you need to install pm2 into your server.

**Comment by shlomi93** [comment]
Another solution which is basically the same lol

ecosystem.config.js:
```
module.exports = {
  apps: [

    {
      name: "my_program",
      script: "path/ to/file.js",

      env: {
        NODE_ENV: "development"
      },
      env_production: {
        NODE_ENV: "production",
        NITRO_PORT: "3560",
      }
    },
  ],
}

```

then
`pm2 start path/to/ecosystem.config.js --env production`
